.github/workflows/pypi.yml

```diff
@@ -0,0 +1,34 @@
+name: Publish to PyPI
```

Comment:
Do you realize that this is a separate workflow that is completely detached from running tests? It will run in parallel to `ci.yaml` regardless of the tests outcome (even before tests complete).

Reply:
I'm assuming this will only happen once merged to master. Therefore, tests should already have been run in the PR before merging. Am I missing something in that logic?

Reply:
Well, that is theoretically okay for as long as you have lockfiles in place. To achieve reproducible builds, I recommend using pip-tools' pairs of "requirements.in + requirements.txt" (dependabot supports this, by the way).